packaging_automation/common_tool_methods.py

```diff
@@ -321,19 +321,20 @@ def prepend_line_in_file(file: str, match_regex: str, append_str: str) -> bool:
 def is_tag_on_branch(tag_name: str, branch_name: str):
     g = git.Git(os.getcwd())
     try:
-        branches_str = g.execute(["git", "branch", "--contains", f"tags/{tag_name}"])
-        branches = remove_prefix(branches_str, "*").split("\n")
-        print("Branches str:" + branches_str)
-        if len(branches) > 0:
-            for branch in branches:
-                if branch.strip() == branch_name:
+        tags = g.execute(["git", "tag", "--merged", f"{branch_name}"])
+        tags = tags.split()
+        print("Tags:", tags)
+        if len(tags) > 0:
+            for tag in tags:
+                if tag.strip() == tag_name:
```

Comment:
Effectively, `git branch --contains` & `git tag --merged` should both work to check if given tag is on a certain branch.

I know you already mentioned that "it was failing", but can you also share the reason behind ?

Reply:
You are right it does not fail until recently but started failing recently. Somehow it does not work
![image](https://user-images.githubusercontent.com/1596331/144811942-38ebcd1a-37f5-4cc4-b4c7-60f90a919429.png)

Reply:
I'm actually wondering the part "`Somehow it does not work`" because this shouldn't be the way we do the changes in this repo, can you please do some research about "why" ?

We should have a better reasoning than "tested & worked" approach.

Reply:
I'm actually wondering the part "`Somehow it does not work`", because this shouldn't be the way we do the changes in this repo, can you please do some research about "why" ?

We should have a better reasoning than "tested & worked" approach.

Reply:
Actually I used the approach which is being used in checklist as above
$ git rev-parse tags/vX.Y.Z
<commitId>
$ git branch --contains <commitId>
release-X.Y
Then I found a stackoverflow post that I can handle it with just one command
https://stackoverflow.com/questions/15806448/git-how-to-find-out-on-which-branch-a-tag-is
![image](https://user-images.githubusercontent.com/1596331/144815706-c15ecffa-8689-406c-b8dd-a98cc88939f0.png)
However when it fails, I deep-dived and found out that the command to query tags is git tag and changed it to this one. 
As you see the documentation,

Reply:
I believe, using `git branch -a --contains tags/<tag>` should also work, as in the link you shared.

Reply:
I believe, using `git branch -a --contains tags/<tag>` should also work, as in the link you shared.

Reply:
I believe, using git branch -a --contains tags/<tag> should also work, as in the link you shared.

Reply:
I've found out that there were another issue about removing * symbol from current branch so can not match the current branch

Reply:
Now fixed without changing the command. As I said before there was an issue about parsing * sign.